utils/dataset_merging/merge_lerobot_dataset.py

```diff
@@ -888,6 +890,255 @@ def pad_parquet_data(source_path, target_path, original_dim=14, target_dim=18):
+            success = True
+            while success:
+                success, _ = cap.read()
+                if success:
+                    frame_count += 1
+        
+        cap.release()
+        return frame_count
+    except ImportError:
+        print("警告: 未安装OpenCV，无法计算视频帧数 (Warning: OpenCV not installed, cannot count video frames)")
+        return 0
+    except Exception as e:
+        print(f"计算视频帧数时出错: {e} (Error counting video frames: {e})")
+        return 0
+
+def copy_images(source_folders, output_folder, episode_mapping, default_fps=20, fps=None):
+    """
+    从源文件夹复制图像文件到输出文件夹，保持正确的索引和结构，并验证图像数量与视频帧数匹配
+    (Copy image files from source folders to output folder, maintaining correct indices and structure,
+    and verify that image count matches video frame count)
+
+    Args:
+        source_folders (list): 源数据集文件夹路径列表 (List of source dataset folder paths)
+        output_folder (str): 输出文件夹路径 (Output folder path)
+        episode_mapping (list): 包含(旧文件夹,旧索引,新索引)元组的列表
+                               (List of tuples containing (old_folder, old_index, new_index))
+    
+    Returns:
+        dict: 验证结果，包含每个episode的预期帧数和实际图像数量
+              (Validation results containing expected frame count and actual image count for each episode)
+    """
+    if fps is None:
+        info_path = os.path.join(source_folders[0], "meta", "info.json")
+        if os.path.exists(info_path):
+            with open(info_path) as f:
+                info = json.load(f)
+                fps = info.get(
+                    "fps", default_fps
+                )  # 使用变量替代硬编码的20 (Use variable instead of hardcoded 20)
+        else:
+            fps = default_fps  # 使用变量替代硬编码的20 (Use variable instead of hardcoded 20)
+
+    print(f"使用FPS={fps} (Using FPS={fps})")
+    # 创建输出图像目录 (Create output image directory)
+    os.makedirs(os.path.join(output_folder, "images"), exist_ok=True)
+    
+    # 跟踪验证结果 (Track validation results)
+    validation_results = {}
+    total_copied = 0
+    
+    # 获取视频路径模板和视频键 (Get video path template and video keys)
+    info_path = os.path.join(source_folders[0], "meta", "info.json")
+    with open(info_path) as f:
+        info = json.load(f)
+    
+    video_path_template = info["video_path"]
+    image_keys = []
+    
+    for feature_name, feature_info in info["features"].items():
+        if feature_info.get("dtype") == "video":
+            image_keys.append(feature_name)
+
+    print(f"Found video/image keys: {image_keys}")
+    print("Starting to copy and validate images...")
+    
+    for old_folder, old_index, new_index in episode_mapping:
+        # 从episodes.jsonl获取预期的帧数 (Get expected frame count from episodes.jsonl)
+        episode_file = os.path.join(old_folder, "meta", "episodes.jsonl")
+        expected_frames = 0
+        if os.path.exists(episode_file):
+            episodes = load_jsonl(episode_file)
+            episode_data = next((ep for ep in episodes if ep["episode_index"] == old_index), None)
```

Comment:
It's time-consuming to find `episode_data` by iterating the whole episodes in a loop.
```suggestion
    episode_file_mapping = {}
    for old_folder, old_index, new_index in episode_mapping:
        # 从episodes.jsonl获取预期的帧数 (Get expected frame count from episodes.jsonl)
        episode_file = os.path.join(old_folder, "meta", "episodes.jsonl")
        expected_frames = 0
        if os.path.exists(episode_file):
            if episode_file not in episode_file_mapping:
                episodes = load_jsonl(episode_file)
                episodes = {ep["episode_index"]: ep for ep in episodes}
                episode_file_mapping[episode_file] = episodes
            episode_data = episode_file_mapping[episode_file].get(old_index, None)
```